src/amber/controllers/base.cr

```diff
@@ -0,0 +1,18 @@
+require "http"
+
+module Amber
+  module Controllers
+    class Base
+      property request : HTTP::Request | Nil = nil
```

Comment:
this is just a matter of semantics / preference, but these could be shortened to `property request : HTTP::Request?`